### DISCUSSION TITLE
How do you use the GridPostArchive component

### DISCUSSION THREAD

**Discussion by Auroraboreali-s** [flyntwp/flynt#387] [Q&A] [answered]
Does anyone uses the GridPostArchive component? I was trying to add it to the pageCompontents.php to be able to use it on a page and every time I get the following error:

`Fatal error: Uncaught Error: Call to undefined function Flynt\Components\GridPostsArchive\getACFLayout() in C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-content\themes\flynt-v1.4.0\inc\fieldGroups\pageComponents.php:28 Stack trace: #0 C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-includes\class-wp-hook.php(292): Flynt\Utils\FileLoader::{closure}() #1 C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-includes\class-wp-hook.php(316): WP_Hook->apply_filters() #2 C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-includes\plugin.php(484): WP_Hook->do_action() #3 C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-content\themes\flynt-v1.4.0\lib\Init.php(28): do_action() #4 C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-includes\class-wp-hook.php(292): Flynt\Init::loadComponents() #5 C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-includes\class-wp-hook.php(316): WP_Hook->apply_filters() #6 C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-includes\plugin.php(484): WP_Hook->do_action( in C:\Bitnami\wordpress-5.6.2-0\apps\wordpress\htdocs\wp-content\themes\flynt-v1.4.0\inc\fieldGroups\pageComponents.php on line 28`

It's a fresh install, all other compontents work correctly. Am I doing something wrong? Does anyone has any idea what is triggering this error, I'm unable to pin point it at the moment.

**Comment by koukouchris** [comment, marked as answer]
Hi @Auroraboreali-s As you see the error states that there is no `getACFLayout()` function in the component. This component is used only for archive page to showcase all posts. It is rendered on `index.twig` and getting all posts from query, its not build as flexible component. If you wish to make it flexible, you will need to refactor it and add `function getACFLayout` and then get all posts manually or with an acf field.

Example:
```
function getACFLayout()
{
    return [
        'name' => 'GridPostsArchive',
        'label' => 'Grid: Posts Archive',
        'sub_fields' => []
    ];
}
```
